include/photos.php

```diff
@@ -42,16 +45,28 @@ function photo_albums($uid, $update = false) {
 	$key = "photo_albums:".$uid.":".local_user().":".remote_user();
 	$albums = Cache::get($key);
 	if (is_null($albums) OR $update) {
-		/// @todo This query needs to be renewed. It is really slow
-		// At this time we just store the data in the cache
-		$albums = qu("SELECT COUNT(DISTINCT `resource-id`) AS `total`, `album`
-			FROM `photo` USE INDEX (`uid_album_created`)
-			WHERE `uid` = %d  AND `album` != '%s' AND `album` != '%s' $sql_extra
-			GROUP BY `album` ORDER BY `created` DESC",
-			intval($uid),
-			dbesc('Contact Photos'),
-			dbesc(t('Contact Photos'))
-		);
+		if (!Config::get('system', 'no_count', false)) {
+			/// @todo This query needs to be renewed. It is really slow
```

Comment:
Yes this does need really some improvement. My commit (https://github.com/friendica/friendica/pull/3046) does have some down-sides. It doesn't work with wall-uploads or uploads through the api (So in this cases the caches isn't updated and we would have old data) and it only updates the data for the owner (if there would be some cached data for visitors they would be wrong)

BTW: is there a way to get the date of an entry in memcache? One solution would be to compare the dates of the cache entries (if the entry of the album owner is newer than the one of the visitor it should update the cache).  https://github.com/rabuzarus/friendica/commit/40c17cee701aa7b7a3921dc4f7beb35c4c97e15f